packages/dashboard-backend/src/devworkspaceClient/services/airGapSampleApi.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2018-2024 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { api } from '@eclipse-che/common';
+import * as console from 'console';
+import * as fs from 'fs';
+import path from 'path';
+
+import { IAirGapSampleApi } from '@/devworkspaceClient/types';
+import { isLocalRun } from '@/localRun';
+
+// See build/dockerfiles/Dockerfile for the path
+const airGapResourcesDir = '/public/dashboard/devfile-registry/air-gap';
+
+export class AirGapSampleApiService implements IAirGapSampleApi {
+  async list(): Promise<Array<api.IAirGapSample>> {
+    return this.readAirGapIndex();
+  }
+
+  async downloadProject(name: string): Promise<api.IStreamedFile> {
+    const sample = this.readAirGapIndex().find(sample => sample.displayName === name);
+    if (sample) {
+      return this.download(sample.project?.zip?.filename);
+    }
+
+    console.error(`Sample not found: ${name} `);
+    throw new Error(`Sample not found`);
+  }
+
+  async downloadDevfile(name: string): Promise<api.IStreamedFile> {
+    const sample = this.readAirGapIndex().find(sample => sample.displayName === name);
+    if (sample) {
+      return this.download(sample.devfile?.filename);
+    }
+
+    console.error(`Sample not found: ${name} `);
+    throw new Error(`Sample not found`);
+  }
+
+  getAirGapResourcesDir(): string {
+    return isLocalRun()
+      ? path.join(
+          __dirname,
+          '../../../dashboard-frontend/lib/public/dashboard/devfile-registry/air-gap',
+        )
+      : airGapResourcesDir;
+  }
+
+  private getAirGapIndexFilePath(): string {
+    return path.join(this.getAirGapResourcesDir(), 'index.json');
+  }
+
+  private readAirGapIndex(): Array<api.IAirGapSample> {
+    const airGapIndexFilePath = this.getAirGapIndexFilePath();
+    if (!fs.existsSync(airGapIndexFilePath)) {
```

Comment:
I don't think it's necessary to do all these synchronous `fs` operations on every request, because the file path doesn't change from one request to another. I'd do it once at the class instantiation.